src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -83,6 +113,27 @@ private CompletableFutures() {
         });
   }
 
+  private static <T> void setupFailFast(
+      FailFast failFast, CompletableFuture<? extends T>[] all, CompletableFuture<Void> allOf) {
+    ExecutionMetadata executionMeta = new ExecutionMetadata(all);
+    for (int i = 0; i < all.length; i++) {
+      all[i].whenComplete(
+          (result, throwable) -> {
+            if (!allOf.isDone()
+                && throwable != null
+                && failFast.failFast(throwable, executionMeta)) {
+              allOf.completeExceptionally(failFast.withThrowable(throwable));
```

Comment:
This could be moved into the FailFast interface - something like `void FastFail.cancelParent(allOf, throwable)` - in case people want to complete it with a value instead of an exception

Reply:
Similar as https://github.com/spotify/completable-futures/pull/68#discussion_r419284692. Allowing people to complete with a default instead than exceptionally, would make this implementation a `ReturnFast` instead of `FailFast` (which is bigger than what this PR intended to do). 

One cannot easily constrain the actions that can be executed in `allOf` inside `cancelParent`, which can lead to API misusage. 

I think the scenario you describe could be implemented by doing: ```
Stream.of(cp1, cp2)
              .collect(CompletableFutures.joinList(new FailFastWithThrowable(WantedException.class)))
              .handle((result, throwable) -> {
                if (throwable instanceof WantedException){
                  return defaultResponse;
                }
                return result;
              });
```

What do you think?